crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -156,25 +166,15 @@ impl CallContractCommand {
 		}
 
 		// Resolve path.
-		let contract_path = match self.path.as_ref() {
-			None => {
-				let path = Some(PathBuf::from("./"));
-				if has_contract_been_built(path.as_deref()) {
-					self.path = path;
-				} else {
-					// Prompt for path.
-					let input_path: String = cli
-						.input("Where is your project located?")
-						.placeholder("./")
-						.default_input("./")
-						.interact()?;
-					self.path = Some(PathBuf::from(input_path));
-				}
-
-				self.path.as_ref().unwrap()
-			},
-			Some(p) => p,
-		};
+		let contract_path = self.path.get_or_insert_with(|| {
+			let input_path: String = cli
+				.input("Where is your project or metadata file located?")
+				.placeholder("./")
+				.default_input("./")
+				.interact()
+				.unwrap();
```

Comment:
Reason for the unwrap? Should we not check it and return an error, or at the very least use .expect() to provide some indication as to why it failed?